python/dolfin/function/functionspace.py

```diff
@@ -5,43 +5,50 @@
 #
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 
+import typing
+
 import ufl
 from dolfin import cpp, jit
 from dolfin.fem import dofmap
 
 
+class ElementDef(typing.NamedTuple):
+    """Data for representing a finite element"""
+    family: str
+    degree: int
+    form_degree: typing.Optional[int] = None  # noqa
+
+
 class FunctionSpace(ufl.FunctionSpace):
```

Comment:
I am not sure why `dolfin.function.FunctionSpace` should inherit from `ufl.FunctionSpace`. 
Its not directly related to this issue, but better think about it. We should probably eliminate inheritance wherever possible.

Reply:
Agree it should be considered (as a separate issue).